HW_5/app.py

```diff
@@ -0,0 +1,71 @@
+'''Необходимо создать API для управления списком задач. Каждая задача должна содержать заголовок и описание. Для каждой задачи должна быть возможность указать статус (выполнена/не выполнена).
+API должен содержать следующие конечные точки:
+— GET /tasks — возвращает список всех задач.
+— GET /tasks/{id} — возвращает задачу с указанным идентификатором.
+— POST /tasks — добавляет новую задачу.
+— PUT /tasks/{id} — обновляет задачу с указанным идентификатором.
+— DELETE /tasks/{id} — удаляет задачу с указанным идентификатором.
+Для каждой конечной точки необходимо проводить валидацию данных запроса и ответа. Для этого использовать библиотеку Pydantic.'''
+
+from fastapi import FastAPI, HTTPException, Path
+from pydantic import BaseModel
+from typing import List, Optional
+
+app = FastAPI()
+
+class TaskBase(BaseModel):
+    title: str
+    description: str
+    status: bool = False
+
+class TaskCreate(TaskBase):
+    pass
+
+class TaskUpdate(TaskBase):
+    pass
+
+class Task(TaskBase):
+    id: int
+
+tasks_db = []
+task_id_counter = 1
+
+# Получение списка всех задач
+@app.get("/tasks", response_model=List[Task])
+def read_tasks():
+    return tasks_db
+
+# Получение задачи по идентификатору
+@app.get("/tasks/{task_id}", response_model=Task)
+def read_task(task_id: int = Path(..., title="The ID of the task")):
+    task = next((t for t in tasks_db if t["id"] == task_id), None)
```

Comment:
А вы пытались получить конкретную задачу? У вас тут ошибка, которая не даст это сделать
```suggestion
    task = next((t for t in tasks_db if t.id == task_id), None)
```